api/src/middleware/assessmentsDummyRouter.ts

```diff
@@ -0,0 +1,291 @@
+      );
+    } catch (error) {
+      next(error);
+      return;
+    }
+
+    res.status(201).json(itemEnvelope(insertedAssessmentSubmissionsRow));
+  }
+);
+
+assessmentsDummyRouter.get(
+  '/makeFacilitator/:programId/:participantId',
+  async (req, res, next) => {
+    const { programId, participantId } = req.params;
+    // a role ID of 2 corresponds to a Facilitator
+    const roleId = 2;
+
+    const programIdParsed = Number(programId);
+    const participantIdParsed = Number(participantId);
+    if (
+      !Number.isInteger(programIdParsed) ||
+      programIdParsed < 1 ||
+      !Number.isInteger(participantIdParsed) ||
+      participantIdParsed < 1
+    ) {
+      next(
+        new BadRequestError(
+          `"${programIdParsed}""${participantIdParsed}"  is not a valid program id .`
+        )
+      );
+      next(
+        new BadRequestError(
+          `"${participantIdParsed}"  is not a valid participant id .`
+        )
+      );
+      return;
+    }
+
+    let insertedProgramFacilitatorsRow: ProgramParticipantsRow;
+    try {
+      insertedProgramFacilitatorsRow = await insertToProgramParticipants(
+        participantIdParsed,
+        programIdParsed,
+        roleId
+      );
+    } catch (error) {
+      next(error);
+      return;
+    }
+    res.status(201).json(itemEnvelope(insertedProgramFacilitatorsRow));
+  }
+);
+interface AssessmentStartRow {
+  id: number;
+  assessment_id?: number;
+  principal_id?: number;
+}
+assessmentsDummyRouter.get(
+  '/startAssessment/:assessmentId/:participantId',
+  async (req, res, next) => {
+    const { assessmentId, participantId } = req.params;
+
+    const participantIdParsed = Number(participantId);
+    const assessmentIdParsed = Number(assessmentId);
+
+    if (
+      !Number.isInteger(participantIdParsed) ||
+      participantIdParsed < 1 ||
+      !Number.isInteger(assessmentIdParsed) ||
+      assessmentIdParsed < 1
+    ) {
+      next(
+        new BadRequestError(
+          `"${participantIdParsed}" "${assessmentIdParsed}" is not a valid participant id or assessment id.`
+        )
+      );
+
+      return;
+    }
+
+    const dummyAssessmentSubmissionData = [
+      {
+        assessmentId: assessmentIdParsed,
+        principalId: participantIdParsed,
+        assessmentSubmissionStateId: 7, // graded
+        score: 10,
+        openedAt: '2023-02-09 12:00:00',
+        submittedAt: '2023-02-09 13:23:45',
+        responses: [{}],
+      },
+    ];
+
+    let insertedAssessmentSubmissionsRow: AssessmentStartRow;
+    try {
+      insertedAssessmentSubmissionsRow = await insertToAssessmentSubmissions(
+        dummyAssessmentSubmissionData[0].assessmentId,
+        dummyAssessmentSubmissionData[0].principalId,
+        dummyAssessmentSubmissionData[0].assessmentSubmissionStateId,
+        dummyAssessmentSubmissionData[0].score,
+        dummyAssessmentSubmissionData[0].openedAt,
+        dummyAssessmentSubmissionData[0].submittedAt,
+        []
```

Comment:
Why are you passing an empty array here as a parameter? On line 268, you should already be defining an empty array, so you should be referencing the "responses" member here.

Reply:
Because I am getting an error when I'm adding  dummyAssessmentSubmissionData.responses

Reply:
See below conversation for resolution for this issue.